### DISCUSSION TITLE
Native (host) and LH LVM disk speeds (differing speed on same physical disk, with LVM)

### DISCUSSION THREAD

**Discussion by voarsh2** [longhorn/longhorn#4186] [Q&A]
Hi.
I'm looking to upgrade to 10Gbe network speeds, but I'm not sure realistically how much it'll help with my current setup (in terms of replica rebuilding times).

I run the following:
```
dd if=/dev/zero of=/tmp/output conv=fdatasync bs=384k count=1k; rm -f /tmp/output
```
I get:
402653184 bytes (403 MB, 384 MiB) copied, 1.44383 s, 279 MB/s - there's also probably some caching, but with the x3 disks at RAID 0 I can expect over 300 MB/s
- Usually it is around 500MB/s

Each node has x3 1TB HDD disks in RAID 0.

In the LH LVM (on the same physical disk as the host (in tmp):
```
root@worker2:/home/reese# dd if=/dev/zero of=/mnt/longhorn1/output1 conv=fdatasync bs=384k count=1k; rm -f /mnt/longhorn1/output1
1024+0 records in
1024+0 records out
402653184 bytes (403 MB, 384 MiB) copied, 13.0525 s, 30.8 MB/s
```

Now, for writing inside a LH volume with just 1 replica:
```
dd if=/dev/zero of=/speed/output conv=fdatasync bs=384k count=1k; rm -f /speed/output
1024+0 records in
1024+0 records out
402653184 bytes (403 MB, 384 MiB) copied, 51.1291 s, 7.9 MB/s
```
--- Edit
After there is no rebuilding on the cluster I ran a test in a 1 replica volume:
```
dd if=/dev/zero of=/speed/output conv=fdatasync bs=384k count=1k; rm -f /speed/output
1024+0 records in
1024+0 records out
402653184 bytes (403 MB, 384 MiB) copied, 11.3768 s, 35.4 MB/s
```
--- Edit 2
I created a new mounted disk (on the same physical disk), not a LVM though, with one replica, I get quite different results:
```
dd if=/dev/zero of=/speed/output conv=fdatasync bs=384k count=1k; rm -f /speed/output
1024+0 records in
1024+0 records out
402653184 bytes (403 MB, 384 MiB) copied, 4.90035 s, 82.2 MB/s
root@ubuntu-speed-6b474955ff-pjcxr:/# dd if=/dev/zero of=/speed/output conv=fdatasync bs=384k count=1k; rm -f /speed/output
1024+0 records in
1024+0 records out
402653184 bytes (403 MB, 384 MiB) copied, 4.414 s, 91.2 MB/s
root@ubuntu-speed-6b474955ff-pjcxr:/# dd if=/dev/zero of=/speed/output conv=fdatasync bs=384k count=1k; rm -f /speed/output
1024+0 records in
1024+0 records out
402653184 bytes (403 MB, 384 MiB) copied, 3.76827 s, 107 MB/s
root@ubuntu-speed-6b474955ff-pjcxr:/# dd if=/dev/zero of=/speed/output conv=fdatasync bs=384k count=1k; rm -f /speed/output
1024+0 records in
1024+0 records out
402653184 bytes (403 MB, 384 MiB) copied, 3.97279 s, 101 MB/s
```
And with two replica's:
```
402653184 bytes (403 MB, 384 MiB) copied, 6.22419 s, 64.7 MB/s
root@ubuntu-speed-6b474955ff-pjcxr:/# dd if=/dev/zero of=/speed/output conv=fdatasync bs=384k count=1k; rm -f /speed/output
1024+0 records in
1024+0 records out
402653184 bytes (403 MB, 384 MiB) copied, 7.25066 s, 55.5 MB/s
root@ubuntu-speed-6b474955ff-pjcxr:/# dd if=/dev/zero of=/speed/output conv=fdatasync bs=384k count=1k; rm -f /speed/output
1024+0 records in
1024+0 records out
402653184 bytes (403 MB, 384 MiB) copied, 6.47216 s, 62.2 MB/s
```

Some volumes are rebuilding, so I expect some performance loss, is this right? writing to the same physical disk, I am getting vastly different speeds. Also, not to mention I am using 1Gbe at the moment, so the max network speed I can expect is around 100MB/s... with 3 nodes, I usually see around 30 MB/s read/write, so I suppose it's reached the network bandwidth?

FYI, the performance hit when actually writing data inside a LH volume is probably going to be worse (see above) than writing a file inside the LVM on the same physical disk.... so something looks amiss if you ask me. I would expect writing to the LVM to give me the same result as /tmp which is the same disk...... but writing inside a LH volume (with a minimum of 2 replica's) to be different....

Extra:
![image](https://user-images.githubusercontent.com/60437695/176210176-a5030f71-c87e-4ae5-9ee4-9668e68e0b09.png)
![image](https://user-images.githubusercontent.com/60437695/176210473-5cc4139a-d37f-431f-a274-8361d1d40f64.png)

**Comment by derekbit** [comment]
Currently, we only have the performance test of Longhorn on SSD.
https://github.com/longhorn/longhorn/wiki/Performance-Benchmark

HDD is not friendly to Longhorn.

**Reply by voarsh2** [reply]
Yes I've seen this mentioned many times, but LH is not friendly for logging, big data, lack of trim, large growing volumes you can't cull...., SSD is a bit on the expensive side for big data... and doesn't answer my question about why LVM is slower......
